src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -129,57 +132,57 @@ public void Write(BuildEventArgs e)
         private void WriteCore(BuildEventArgs e)
         {
             // the cases are ordered by most used first for performance
-            if (e is BuildMessageEventArgs)
+            if (e is BuildMessageEventArgs buildMessage)
```

Comment:
Could this whole thing be turned into a switch statement? But also: 👍

Reply:
I’ll need to measure which is faster

Reply:
Indeed, a switch statement is slightly faster: 
https://github.com/KirillOsenkov/Benchmarks/blob/7d6d9323374a58994e5dfc3fb6e01c6bfda3f3c7/Benchmarks/Tests/SwitchStatement.cs#L7-L8

Reply:
Switched to a switch statement ;P

Reply:
I did something like that in several places in Evaluator at one point. It ran a lot. That alone saved like 2% of evaluate time.